src/predict_request_validation_utils.cpp

```diff
@@ -848,38 +900,45 @@ Status RequestValidator<RequestType, InputTensorType, IteratorType, ShapeType>::
         }
         const Dimension& batchSize = inputInfo->getShape()[batchIndex.value()];
         Mode shapeMode = getShapeMode(shapeInfo, name);
-        if (checkIfNativeFileFormatUsed(proto, name)) {
-            status = validateNumberOfBinaryInputShapeDimensions(proto);
-            if (!status.ok())
-                return status;
-
-            if (shouldValidateBinaryBatchSizeMismatch(request)) {
-                status = checkBinaryBatchSizeMismatch(proto, batchSize, finalStatus, batchingMode, shapeMode);
-                if (!status.ok())
-                    return status;
-            } else if (batchSize != 1) {
-                SPDLOG_DEBUG("When the image is placed in raw_inputs_contents batch size cannot be bigger than 1.");
-                return StatusCode::INVALID_BATCH_SIZE;
-            }
 
-            continue;
+        if (requiresProcessing(proto)) {
+            const auto processingHint = inputInfo->getProcessingHint();
+            if (processingHint == TensorInfo::ProcessingHint::STRING_1D_U8) {
+                SPDLOG_DEBUG("[servable name: {} version: {}] Validating request containing 1D string input: name: {}; batch size: {}",
+                    servableName, servableVersion, name, batchSize.toString());
+                RETURN_IF_ERR(validateNumberOfBinaryInputShapeDimensions(proto));
+                continue;
+            } else if (processingHint == TensorInfo::ProcessingHint::STRING_2D_U8) {
+                SPDLOG_DEBUG("[servable name: {} version: {}] Validating request containing 2D string input: name: {}; batch size: {}",
+                    servableName, servableVersion, name, batchSize.toString());
+                RETURN_IF_ERR(validateNumberOfBinaryInputShapeDimensions(proto));
+                RETURN_IF_ERR(checkBinaryBatchSizeMismatch(proto, batchSize, finalStatus, batchingMode, shapeMode));
+                RETURN_IF_ERR(checkStringShapeMismatch(proto, *inputInfo, finalStatus, batchingMode, shapeMode));
```

Comment:
NLP models with 1d u8 (MUSE only for now) do not encode batch nor shape in ov::Tensor metadata. The models require 1d u8 array of bytes with batch/string lengths encoded inside data of the tensor.

Since we convert from TFS/KServe proto into the 1d u8 format we have no information what batch/lengths are supported by the model. All we do is convert from proto into the 1d u8 format and rely on validation on custom extension side (during async infer of OV model).